dbms/src/Common/Exception.h

```diff
@@ -349,4 +349,14 @@ constexpr auto maybeStringLiteralExpr(const std::string_view sv)
         }                                                              \
     } while (false)
 
+#define CATCH_AND_LOG(stmt, log)                              \
```

Comment:
If `CATCH_AND_LOG` is only used by `consumeResource`, how about letting `consumeResource` not throw an error?

Reply:
done. it will return bool to indicate succ or not